include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -408,19 +412,17 @@ def AndRasUInt : Pat <
   (MoveNameHint $old, (AndRPrimOp $x)),
   [(KnownWidth $x), (IntTypes $x)]>;
 
-/// andr(cvt(x:unsigned)) -> 0
-def AndRCvtU : Pat <
-  (AndRPrimOp:$old (CvtPrimOp $x)),
-  (MoveNameHint $old,
-    (NativeCodeCall<"$_builder.create<ConstantOp>($0.getLoc(), cast<IntType>($0.getType()), getIntZerosAttr($0.getType()))"> $old)),
-  [(KnownWidth $x), (UIntTypes $x)]>;
-
-/// andr(cvt(x:signed)) -> andr(x)
-def AndRCvtS : Pat <
-  (AndRPrimOp:$old (CvtPrimOp $x)),
-  (MoveNameHint $old, (AndRPrimOp $x)),
-  [(KnownWidth $x), (SIntTypes $x)]>;
+/// andr(cat(0, x)) -> 0
+def AndRCatZeroL : Pat <
+  (AndRPrimOp:$old (CatPrimOp $z, $x)),
+  (NativeCodeCall<"$_builder.create<ConstantOp>($0.getLoc(), cast<IntType>($0.getType()), getIntZerosAttr($0.getType()))"> $old),
+  [(ZeroConstantOp $z)]>;
 
+/// andr(cat(x, 0)) -> 0
+def AndRCatZeroR : Pat <
+  (AndRPrimOp:$old (CatPrimOp $x, $z)),
+  (NativeCodeCall<"$_builder.create<ConstantOp>($0.getLoc(), cast<IntType>($0.getType()), getIntZerosAttr($0.getType()))"> $old),
+  [(ZeroConstantOp $z)]>;
```

Comment:
More generally we can fold into `0` when `z != -1` and also `andr(cat(x, -1)) -> andr(x)`

Reply:
similarly `andr(pad(x:Signed, w)) -> 1` when `w > sizeof(x)`